velox/type/TimestampConversion.h

```diff
@@ -269,10 +269,41 @@ inline Expected<ParsedTimestampWithTimeZone> fromTimestampWithTimezoneString(
   return fromTimestampWithTimezoneString(str.data(), str.size(), parseMode);
 }
 
+/// Converts ParsedTimestampWithTimeZone to Timestamp according to the
+/// timezone-based adjustment. If no timezone information is available
+/// in the first argument, respects the session timezone if configured.
+inline Timestamp fromParsedTimestampWithTimeZone(
+    ParsedTimestampWithTimeZone parsed,
+    const tz::TimeZone* sessionTimeZone) {
```

Comment:
Might be better to implement this function in the source file as it is not simple.